question-servers/elements/pl_answer_panel.py

```diff
@@ -18,5 +18,8 @@ def parse(element_html, element_index, data):
 def grade(element_html, element_index, data):
     return data
 
-def testcase(element_html, element_index, data):
+def test(element_html, element_index, data):
     return data
+
+def file(element_html, element_index, data):
```

Comment:
I think this represents a fundamentally different kind of element "life cycle" phase than things like render. You'll basically never have a component that doesn't render any content, so it makes sense to complain if that function is missing. On the other hand, components with a `file` function that does anything will be the exception rather than the rule, so I think not erroring if it's missing makes sense. Same could be said for phases like `grade`: my file upload component doesn't grade itself, so there's no need to error if that function is missing